src/fragments/lib/datastore/ios/data-access/delete-snippet.mdx

```diff
@@ -36,16 +38,16 @@ A delete can also be achieved by a `type` and its `id`.
 
 <BlockSwitcher>
 
-<Block name="Listener (iOS 11+)">
+<Block name="Async/Await">
 
 ```swift
-Amplify.DataStore.delete(Post.self, withId: "123") {
-    switch $0 {
-    case .success:
-        print("Post deleted!")
-    case .failure(let error):
-        print("Error deleting post - \(error.localizedDescription)")
-    }
+do {
+    try await Amplify.DataStore.delete(Post.self, withIdentifier: "123")
```

Comment:
I'm getting error if I use `withIdentifier` instead of `withId` as:
_`Instance method 'delete(_:withIdentifier:where:)' requires that 'Post' conform to 'ModelIdentifiable'`_
Is it right if I could just modify my test model to conform to ModelIdentifiable ? will customers have to do the same?

Reply:
@lawmicha this has to do with the CPK changes. I wonder where CLI is at with this.